numpy/lib/function_base.py

```diff
@@ -4294,8 +4303,21 @@ def percentile(a, q, axis=None, out=None,
         return r
 
 
-def _percentile(a, q, axis=None, out=None,
-                overwrite_input=False, interpolation='linear', keepdims=False):
+def _quantile_is_valid(q):
+    # avoid expensive reductions, relevant for arrays with < O(1000) elements
+    if q.ndim == 1 and q.size < 10:
+        for i in range(q.size):
```

Comment:
While you're at it, would it be an idea to just remove this special casing? The timing does switch at 10 elements, at ~3us on my machine, and for 2 elements the python loop is 1.2us, while the numpy loop is still ~3us. But why do we care if next we are going to sort a whole array? This optimization is just not worth it.

If we do keep the loop, maybe at least make it do:
```
    for _q in q:
        if not (0.0 <= _q <= 1.0):
            return False
```
(which is faster both for not getting indices and for doing the combined comparison)

p.s. the `count_nonzero` is worth it

Reply:
Was planning to put that in a separate PR